### PR TITLE
[GPU] Skip empty subgraph of If operator

### PR DESCRIPTION
**Pull request by yeonbok**
### Details:
 - Skip subgraph of if operator, if it has only input and output nodes.

### Tickets:
 - 124832

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/program.cpp [hunk 1]
```diff
@@ -161,6 +161,23 @@ program::program(engine& engine_ref,
...
+        if (_is_body_program) {
+            // To skip empty if (condition) subgraph
+            bool can_be_optimized = true;
+            for (auto& node : processing_order) {
+                if (node->is_type<input_layout>()) {
+                    continue;
+                } else if (node->is_type<data>()) {
+                    continue;
+                } else if (node->is_output() && !node->has_fused_primitives() &&
+                      node->get_input_layout(0).data_type == node->get_output_layouts(false)[0].data_type) {
```
Comment (sshlyapn) [inline src/plugins/intel_gpu/src/graph/program.cpp:173]: I think it is worth to check that data format is not changed too

File: src/plugins/intel_gpu/src/graph/impls/common/condition.cpp [hunk 2]
```diff
@@ -41,9 +41,14 @@ struct condition_impl : typed_primitive_impl<condition> {
         auto pred = condition_inst::get_pred_from_memory(instance.pred_memory_ptr(), stream);
         network::ptr executed_net = pred ? instance.get_net_true() : instance.get_net_false();
         auto branch = pred ? instance.get_branch_true() : instance.get_branch_false();
-        executed_net->set_shape_predictor(instance.get_network().get_shape_predictor());
+        bool can_skip_subgraph = branch.inner_program->can_be_optimized();
+        if (can_skip_subgraph)
```
Comment (sshlyapn) [inline src/plugins/intel_gpu/src/graph/impls/common/condition.cpp:45]: Was your intention not to set shape predictor for networks which can be optimized?
```suggestion
        if (!can_skip_subgraph)
```

File: src/plugins/intel_gpu/src/graph/impls/common/condition.cpp [hunk 3]
```diff
@@ -54,25 +59,41 @@ struct condition_impl : typed_primitive_impl<condition> {
...
-                GPU_DEBUG_LOG << "Inner net - Inputs[" << mem_idx << "]" << mem_ptr->get_layout().to_short_string() << std::endl;
+                if (can_skip_subgraph) {
+                    const auto& input_internal_node = branch.inner_program->get_node(input_internal_id);
+                    // look for corresponding result output
+                    const auto& internal_output_id = (*input_internal_node.get_users().begin())->id();
+                    using out_map_type = std::pair<size_t, cldnn::primitive_id>;
+                    auto outer_output = std::find_if(branch.output_map.begin(), branch.output_map.end(), [&](const out_map_type& m) {
+                        return internal_output_id == m.second;
+                    });
+                    instance.set_output_layout(mem_ptr->get_layout());
```
Comment (sshlyapn) [inline src/plugins/intel_gpu/src/graph/impls/common/condition.cpp:76]:
```suggestion
                    instance.set_output_layout(mem_ptr->get_layout(), outer_output->first);
```

File: src/plugins/intel_gpu/src/graph/impls/common/condition.cpp (hunk 2 above)
Comment (yeonbok) [reply src/plugins/intel_gpu/src/graph/impls/common/condition.cpp:45]: that was typo! thanks!

File: src/plugins/intel_gpu/src/graph/program.cpp (hunk 1 above)
Comment (yeonbok) [reply src/plugins/intel_gpu/src/graph/program.cpp]: Applied, thanks

**Review by ahnyoung-paul — APPROVED** [review]

**Review by sshlyapn — APPROVED** [review]

**Pull request merged**